cypress/integration/aaaaaaaaaaaaaaaaaaaaaaaa/add_a_product_to_a_pricelist_schema_and_create_a_new_PLV.js

```diff
@@ -0,0 +1,184 @@
+      .addLine(new PriceListSchemaLine().setProduct(productName1).setStandardPriceSurchargeAmount(surchargeAmount))
+      .apply();
+  });
+
+  it('Expect Product1 has a single Product Price', function() {
+    filterProductPricesByProduct(productName1);
+    cy.expectNumberOfRows(1);
+  });
+
+  it('Expect PLV1 has 2 Product Prices', function() {
+    filterProductPricesByPLV(priceListName, priceListVersionNameSearch1);
+    cy.expectNumberOfRows(2);
+  });
+
+  it('Save initial Product Price data', function() {
+    cy.selectNthRow(0).dblclick();
+    cy.getStringFieldValue('PriceStd').then(val => {
+      originalPriceStd = parseFloat(val);
+    });
+    cy.getStringFieldValue('PriceLimit').then(val => {
+      originalPriceLimit = parseFloat(val);
+    });
+    cy.getStringFieldValue('PriceList').then(val => {
+      originalPriceList = parseFloat(val);
+    });
+    cy.getStringFieldValue('C_UOM_ID').then(val => {
+      originalUOM = val;
+    });
+    cy.getStringFieldValue('C_TaxCategory_ID').then(val => {
+      originalTaxCategory = val;
+    });
+  });
+});
+
+describe('Create new Price List Version using the Price List Schema', function() {
+  it('Add the Schema PriceList Version', function() {
+    cy.visitWindow('540321', priceListID);
+    PriceList.applyPriceListVersion(
+      new PriceListVersion()
+        .setName(priceListSchemaVersionName)
+        .setValidFrom(priceListVersion2ValidFrom)
+        .setDiscountSchema(priceListSchemaName)
+        .setBasisPricelistVersion(priceListVersionName)
+    );
+    cy.expectNumberOfRows(2);
+  });
+
+  it('Run action "Create Price List"', function() {
+    // the PriceListSchema PLV should always be the first, as its date is the biggest and the default sorting is by ValidFrom, desc
+    cy.selectNthRow(0);
+    cy.executeHeaderAction('M_PriceList_Create');
+    cy.waitForSaveIndicator();
+  });
+
+  it('Expect Product1 has 2 Product Prices', function() {
+    filterProductPricesByProduct(productName1);
+    cy.expectNumberOfRows(2);
+  });
+
+  it('Expect PLV1 has 2 Product Prices', function() {
+    filterProductPricesByPLV(priceListName, priceListVersionNameSearch1);
+    cy.expectNumberOfRows(2);
+  });
+
+  it('Expect PLV2 has 1 Product Price', function() {
+    filterProductPricesByPLV(priceListName, priceListVersionNameSearch2);
+    cy.expectNumberOfRows(1);
+  });
+
+  it('Check the new Product Price', function() {
+    cy.selectNthRow(0).dblclick();
+    cy.getStringFieldValue('PriceStd').should(val => {
+      expect(parseFloat(val)).to.be.closeTo(originalPriceStd + surchargeAmount, 0.01);
+    });
+    cy.getStringFieldValue('PriceLimit').should(val => {
+      expect(parseFloat(val)).to.be.closeTo(originalPriceLimit, 0.01);
+    });
+    cy.getStringFieldValue('PriceList').should(val => {
+      expect(parseFloat(val)).to.be.closeTo(originalPriceList, 0.01);
+    });
+    cy.getStringFieldValue('M_Product_ID').should('contains', productName1);
```

Comment:
Typo ?

Reply:
what typo? i don't know what you're referring to :-?